geode-core/src/integrationTest/java/org/apache/geode/internal/cache/DiskStoreImplIntegrationTest.java

```diff
@@ -238,6 +238,60 @@ public void givenDiskStoreIsCreatedWithAtLeastOneDefaultSize_whenDataIsStored_th
         ((DiskStoreImpl) diskStore).getDiskUsagePercentage(), 0);
   }
 
+  @Test
+  public void whenMaxOplogFileDoesNotFitInDir_thenNextDirIsSelected() throws Exception {
+
+    File[] diskDirs = new File[3];
+    diskDirs[0] = temporaryDirectory.newFolder("dir1");
+    diskDirs[1] = temporaryDirectory.newFolder("dir2");
+    diskDirs[2] = temporaryDirectory.newFolder("dir3");
+    final int NUM_ENTRIES = 450000;
+    int[] diskDirSizes;
+
+    cache = createCache();
+    diskDirSizes = new int[] {10, 5, 10};
+
+    cache.createDiskStoreFactory().setMaxOplogSize(2).setDiskDirsAndSizes(diskDirs, diskDirSizes)
+        .create(DISK_STORE_NAME);
+    Region region = cache.<String, String>createRegionFactory(RegionShortcut.PARTITION_PERSISTENT)
+        .setDiskStoreName(DISK_STORE_NAME).create(REGION_NAME);
+    DiskStore diskStore = cache.findDiskStore(DISK_STORE_NAME);
+
+    putEntries(region, NUM_ENTRIES);
+
+    File[] dirs = diskStore.getDiskDirs();
+
+    assertThat(oplogFileIsInDir(1, dirs[0])).isTrue();
+    assertThat(oplogFileIsInDir(2, dirs[1])).isTrue();
+    assertThat(oplogFileIsInDir(3, dirs[2])).isTrue();
+
+    assertThat(oplogFileIsInDir(4, dirs[0])).isTrue();
+    assertThat(oplogFileIsInDir(5, dirs[1])).isTrue();
+    assertThat(oplogFileIsInDir(6, dirs[2])).isTrue();
+
+    assertThat(oplogFileIsInDir(7, dirs[0])).isTrue();
+    assertThat(oplogFileIsInDir(8, dirs[1])).isFalse();
```

Comment:
I think I understand what this test is doing now... It's just showing that due to the size difference in dir2, that we skipped over it and created the oplog in dir3 instead.  Is there a way to show this with less true/false checks?  Maybe reduce the size of the failing directory or number or directories?  Or perhaps a comment about why 8 should be false for dir[1] and present in dir[2].

Reply:
You are right, maybe the test is not so easy to understand as I thought, so I have added an explanation about what the test is doing and the expected result. Thanks!